tests/components/bluetooth_le_tracker/test_device_tracker.py

```diff
@@ -0,0 +1,62 @@
+"""Test Bluetooth LE device tracker."""
+
+from datetime import timedelta
+from unittest.mock import Mock, patch
+
+from homeassistant.components.bluetooth_le_tracker import device_tracker
+from homeassistant.components.device_tracker.const import (
+    CONF_SCAN_INTERVAL,
+    CONF_TRACK_NEW,
+    SOURCE_TYPE_BLUETOOTH_LE,
+)
+from homeassistant.const import ATTR_NOW, EVENT_TIME_CHANGED
+from homeassistant.util import dt as dt_util
+
+from tests.common import get_test_home_assistant
+
+
+def test_preserve_new_tracked_device_name():
+    """Test preserving tracked device name across new seens."""
+
+    address = "DE:AD:BE:EF:13:37"
+    name = "Mock device name"
+
+    hass = get_test_home_assistant()
+    hass.start()
+
+    mock_see = Mock()
+
+    with patch(
+        "homeassistant.components."
+        "bluetooth_le_tracker.device_tracker.pygatt.GATTToolBackend"
+    ) as mock_backend, patch.object(device_tracker, "MIN_SEEN_NEW", 3):
+
+        # Return with name when seen first time
+        device = {"address": address, "name": name}
+        mock_backend.return_value.scan.return_value = [device]
+
+        config = {CONF_SCAN_INTERVAL: timedelta(minutes=1), CONF_TRACK_NEW: True}
+        res = device_tracker.setup_scanner(hass, config, mock_see)
```

Comment:
Please instead use the core interfaces to set up the component and platform and assert the entity state via the state machine at the end of the test. We should avoid setting up the platform directly or interfering with it. Doing so will make the tests more robust.

Reply:
Done